### ISSUE TITLE
Column.NumberFormat is not loaded correctly

### ISSUE THREAD

**Issue by GeorgDangl** [ClosedXML/ClosedXML#961]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

0.93

**What is the current behavior?**

When a `NumberFormat` is specified for a column, it's not present in the resulting Excel file, but a different number format is set.
For example, this code:
```C#
var sourceColumn = sourceWorkbook.Worksheets.Single().Column(1);
                sourceColumn.Style.NumberFormat.Format = "0,00";
```
Produces this format when viewed in Excel: `#.000`. It's an empty string when opened again with ClosedXML.

**Did this work in previous versions of our tool?  Which versions?**

It worked in versions earlier than 0.93.

## Reproducibility

The following test case shows the problem:

```c#
[Fact]
public void ReadAndWriteColumnNumberFormat()
{
    using (var sourceWorkbook = new XLWorkbook())
    {
        sourceWorkbook.AddWorksheet("MyCalculation");
        var sourceColumn = sourceWorkbook.Worksheets.Single().Column(1);
        sourceColumn.Style.NumberFormat.Format = "0,00";

        using (var memoryStream = new MemoryStream())
        {
            sourceWorkbook.SaveAs(memoryStream);
            memoryStream.Position = 0;

            using (var destWorkbook = new XLWorkbook(memoryStream))
            {
                var column = destWorkbook.Worksheets.Single().Column(1);
                // This test fails, the read value is an empty string
                Assert.Equal("0,00", column.Style.NumberFormat.Format);
            }
        }
    }
}
```

Here's the defined number formats in the resulting `styles.xml`:

```xml
<x:numFmts count="2">
  <x:numFmt numFmtId="0" formatCode="" />
  <x:numFmt numFmtId="165" formatCode="0,00" />
</x:numFmts>
```

So the `numFmtId` is set to `165` and present, but it's not referenced in `sheet1.xml`:

```xml
<x:cols>
  <x:col min="1" max="1" width="9.140625" style="1" customWidth="1" />
</x:cols>
```

I'm not exactly sure what happens. I think the reference should be `style="165"`. It looks like the reference to style `1` gets picked up by excel as some internal, pre-defined style (numbers with thousands separator, in this case).

**Comment by igitur** [comment]
@Pankraty Please have a look at this. `IXLStylized` containers should propage style properties to its child elements (in `IXLStylized.Styles`).

**Comment by igitur** [comment]
@Pankraty Please have a look at this. `IXLStylized` containers should propage style properties to its child elements (in `IXLStylized.Styles`). The new style repository broke this part. I'm a bit hesitant to touch this myself.

**Comment by GeorgDangl** [comment]
Hey there, I've seen the 0.94 release and noticed it contains quite a few improvements and bugfixes. Do you guys have any idea when this issue here could be tackled? Is there any workaround I could apply except by specifying the number formats individually per cell?

**Comment by Pankraty** [comment]
@GeorgDangl I fixed the issue with loading styles applied to whole columns. As for your problem with replacing "0,00" with "#.000", pay attention, that Excel accepts a period as a decimal separator, not a comma.
![image](https://user-images.githubusercontent.com/19576939/51253978-5196eb00-19b9-11e9-924d-147af73b08e2.png)

**Comment by GeorgDangl** [comment]
Hey @Pankraty, thanks for the info!
Can you say when the update will be available on NuGet?

As to the format:
The problem I was describing doesn't depend on the format chosen, I've just used `#.000` as an example. The root of the problem was that whatever format I specify, when I serialize to Excel and then deserialize again, the format is imported as `#.000` when I open the file with Excel, but as empty string when opening with ClosedXML.

This unit test shows the defect. It works with 0.92.1 but fails with 0.94.2:

```csharp
[Fact]
public void ReadAndWriteColumnNumberFormat()
{
    // This problem was introduced with ClosedXML v0.93.0
    using (var sourceWorkbook = new XLWorkbook())
    {
        sourceWorkbook.AddWorksheet("MyCalculation");
        var sourceColumn = sourceWorkbook.Worksheets.Single().Column(1);
        sourceColumn.Style.NumberFormat.Format = "0 \"hours\"";

        using (var memoryStream = new MemoryStream())
        {
            sourceWorkbook.SaveAs(memoryStream);
            memoryStream.Position = 0;

            using (var destWorkbook = new XLWorkbook(memoryStream))
            {
                var column = destWorkbook.Worksheets.Single().Column(1);
                // This test fails, the read value is an empty string
                Assert.Equal("0 \"hours\"", column.Style.NumberFormat.Format);
            }
        }
    }
}
```
In the test, I've set the format to `0 "hours"`, but it doesn't matter which format I chose, it's simply not present when I deserialize it again.

**Comment by Pankraty** [comment]
> The problem I was describing doesn't depend on the format chosen

Yes, I know. I've fixed it in #1135. You can try the artifacts from [this build](https://ci.appveyor.com/project/ClosedXML/closedxml/builds/21662187/job/whnui5i7ikn98xce/artifacts).

When to release the new version of ClosedXML is up to @igitur to decide. He is the boss ;)

**Comment by GeorgDangl** [comment]
Thank you very much, I just ran all tests and everything is green!
